examples/versatileab/src/bin/prefetch-exception-a32.rs

```diff
@@ -0,0 +1,85 @@
+//! Example triggering a prefetch exception.
+
+#![no_std]
+#![no_main]
+
+use core::sync::atomic::{AtomicU32, Ordering};
+use cortex_ar::register::{Ifar, Ifsr};
+use semihosting::println;
+
+// pull in our start-up code
+use versatileab as _;
+
+static COUNTER: AtomicU32 = AtomicU32::new(0);
+
+/// The entry-point to the Rust application.
+///
+/// It is called by the start-up.
+#[no_mangle]
+pub extern "C" fn kmain() -> ! {
+    println!("Hello, this is a prefetch exception example");
+
+    // A BKPT instruction triggers a Prefetch Abort except when Halting debug-mode is enabled.
+    // See p. 2038 of ARMv7-M Architecture Reference Manual
+    unsafe {
+        // trigger an prefetch exception, from A32 (Arm) mode
+        bkpt_from_a32();
+    }
+
+    // this should be impossible because returning from the fault handler will
+    // immediately trigger the fault again.
+
+    unreachable!("should never be here!");
+}
+
+// These functions are written in assembly
+extern "C" {
+    fn bkpt_from_a32();
+}
+
+core::arch::global_asm!(
+    r#"
+    // fn bkpt_from_a32();
+    .arm
+    .global bkpt_from_a32
+    .type bkpt_from_a32, %function
+    bkpt_from_a32:
+        bkpt    #0
+        bx      lr
+    .size bkpt_from_a32, . - bkpt_from_a32
+"#
+);
```

Comment:
unrelated, but this looks like it could fit `cortex-a::asm`/`cortex-r::asm`. Perhaps as an `unsafe` function?

Reply:
Yes, but it doesn't help here because I need to know the *precise* address of the instruction that does the breakpoint.